### ISSUE TITLE
Kubernetes routing loop involving calico-node

### ISSUE THREAD

**Issue by tasuren** [tailscale/tailscale#7594]
### What is the issue?

I tried to use a private network created by Tailscale to build a cluster of k8s (Kubernetes).
Then, Tailscale running on two servers running worker nodes joined to the cluster started to consume an unnatural amount of network (tx 600Mbps) and CPU (80%, total 100%).
Why is this?

Log of tailscaled:
```shell
$ journalctl -fu tailscaled
 3月 18 12:26:21 ... tailscaled[60693]: magicsock: disco: failed to send *disco.CallMeMaybe to 127.3.3.40:7: too many DERP packets queued; dropping
 3月 18 12:26:27 ... tailscaled[60693]: Accept: UDP{100.97.160.100:32927 > 100.112.81.21:4789} 638 ok
 3月 18 12:26:34 ... tailscaled[60693]: monitor: RTM_DELROUTE: src=, dst=fe80::/64, gw=, outif=59, table=254
 3月 18 12:26:34 ... tailscaled[60693]: monitor: RTM_DELROUTE: src=, dst=fe80::/128, gw=, outif=59, table=255
 3月 18 12:26:34 ... tailscaled[60693]: monitor: RTM_DELROUTE: src=, dst=fe80::ecee:eeff:feee:eeee/128, gw=, outif=59, table=255
 3月 18 12:26:34 ... tailscaled[60693]: monitor: RTM_DELROUTE: src=, dst=ff00::/8, gw=, outif=59, table=255
 3月 18 12:26:34 ... tailscaled[60693]: magicsock: endpoints changed: <CP Node IP>:53641 (stun), <CP Node IP>:41641 (stun4localport), 10.96.2.0:41641 (local), 10.244.55.193:41641 (local), 192.168.3.22:41641 (local), [<Some IPv6>]:41641 (local)
 3月 18 12:26:34 ... tailscaled[60693]: magicsock: endpoints changed: <CP Node IP>:35358 (stun), <CP Node IP>:41641 (stun4localport), 10.96.2.0:41641 (local), 10.244.55.193:41641 (local), 192.168.3.22:41641 (local), [<Some IPv6>]:41641 (local)
 3月 18 12:26:37 ... tailscaled[60693]: Accept: UDP{100.112.81.21:47872 > 100.97.160.100:4789} 638 ok out
 3月 18 12:26:37 ... tailscaled[60693]: magicsock: disco: failed to send *disco.CallMeMaybe to 127.3.3.40:7: too many DERP packets queued; dropping
 3月 18 12:26:47 ... tailscaled[60693]: Accept: UDP{100.112.81.21:33641 > 100.97.160.100:4789} 1278 ok out
 3月 18 12:26:57 ... tailscaled[60693]: Accept: UDP{100.97.160.100:32927 > 100.112.81.21:4789} 638 ok
 3月 18 12:26:58 ... tailscaled[60693]: magicsock: endpoints changed: <CP Node IP>:53641 (stun), <CP Node IP>:41641 (stun4localport), 10.96.2.0:41641 (local), 10.244.55.193:41641 (local), 192.168.3.22:41641 (local), [<Some IPv6>]:41641 (local)
```

### Steps to reproduce

1. Prepare k8s' controle-plane node and tailscale.
2. Prepare k8s' worker node and join it to controle-plane node.

### Are there any recent changes that introduced the issue?

After the second worker node is joined, the problem occurs until tailscale is brought down on both nodes or k8s is reset.

### OS

Linux

### OS version

Rocky Linux 9.1

### Tailscale version

1.38.1

### Other software

- k8s (Kubernetes)
- Calico for CNI of k8s (The same phenomenon occurred in Cilium.)

### Bug report

BUG-199680f3bb80e6faa12a6cecd7fcff3fe73ca0d9a45008c8bba874b857e9840c-20230318035813Z-819d637d6cc505ad

**Comment by DentonGentry** [comment]
This sounds like a routing loop.

Are you using the [Tailscale Kubernetes Operator](https://github.com/tailscale/tailscale/issues/502#issuecomment-1414522795), or a sidecar?

**Comment by tasuren** [comment]
I heard that for first time. I'm going to see that.

Also, I found that if calico-node is not running at least one worker node, the problem is solved. (Although the network is only available on the worker node that is running.)

**Comment by K8SLIKE** [comment]
Kubernetes,calico-node Keep restarting？

**Comment by tasuren** [comment]
No restarting.

**Comment by ciffelia** [comment]
I have the same problem. My cluster uses Tailscale for communication between **nodes** (not pods) and Calico (VXLAN) for CNI.

Recently I noticed that Tailscale's CPU usage was frequently spiking and communication between nodes was becoming unstable, so I checked the logs and found the following message.

```
tailscaled[841]: magicsock: disco: node [XXX] d:xxx now using 10.244.x.y:41641
```

Bug report: BUG-9933ac6858348ab1ecd0a30a4e1ea9563e5a2c5da197d8b758c4b1cc51b14436-20230827064054Z-f455a54d6b409056

The CIDR of my cluster's pod network is `10.244.0.0/16`. The fact that Tailscale is trying to communicate with other nodes through `10.244.x.y` suggests that there is a loop between Tailnet and the pod network. In other words, packets sent from one node to another are first encapsulated by Tailscale, then by Calico, then by Tailscale, and so on.

I think this problem can be solved if #1552 is implemented; I just need to exclude `10.244.0.0/16`.
